src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2215,76 +2338,35 @@ public override BoundNode VisitConversion(BoundConversion node)
-                        if (!node.ExplicitCastInCode && operand.Kind == BoundKind.Lambda)
-                        {
-                            var lambda = (BoundLambda)operand;
-                            ReportNullabilityMismatchWithTargetDelegate(operand.Syntax, targetType.GetDelegateType(), lambda.Symbol);
-                        }
-                        break;
-
-                    case ConversionKind.MethodGroup:
-                        if (!node.ExplicitCastInCode)
-                        {
-                            ReportNullabilityMismatchWithTargetDelegate(operand.Syntax, targetType.GetDelegateType(), node.SymbolOpt);
-                        }
-                        break;
+                    reportNullable = true;
                 }
-
-                if (node.ExplicitCastInCode && !node.IsExplicitlyNullable)
+                else if ((targetType.IsReferenceType || IsUnconstrainedTypeParameter(targetType)) &&
+                    resultType?.IsNullable == true)
                 {
-                    bool reportNullable = false;
-                    if (targetType.IsReferenceType && IsUnconstrainedTypeParameter(operandType?.TypeSymbol))
-                    {
-                        reportNullable = true;
-                    }
-                    else if ((targetType.IsReferenceType || IsUnconstrainedTypeParameter(targetType)) &&
-                        (operandType?.IsNullable == true || (operandType is null && operand.IsLiteralNullOrDefault())))
-                    {
-                        reportNullable = true;
-                    }
-                    if (reportNullable)
-                    {
-                        // PROTOTYPE(NullableReferenceTypes): Should not report warning for explicit
-                        // user-defined conversion if the operator is defined from nullable to
-                        // non-nullable. See StaticNullChecking.ExplicitCast_UserDefined.
-                        ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_ConvertingNullableToNonNullable, node.Syntax);
-                    }
-                }
-
-                TypeSymbolWithAnnotations resultType;
-                if (operand.Kind == BoundKind.Literal && (object)operand.Type == null && operand.ConstantValue.IsNull)
-                {
-                    resultType = TypeSymbolWithAnnotations.Create(targetType, true);
-                }
-                else if (node.ConversionKind == ConversionKind.Identity && !node.ExplicitCastInCode)
-                {
-                    resultType = operandType;
+                    reportNullable = true;
                 }
-                else
+                if (reportNullable)
                 {
-                    resultType = InferResultNullability(operand, node.Conversion, targetType, operandType, fromConversionNode: true);
+                    ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_ConvertingNullableToNonNullable, node.Syntax);
```

Comment:
>WRN_ConvertingNullableToNonNullable [](start = 66, length = 35)

This is a "W" warning, right?

Reply:
Yes, that's the one: `WRN_ConvertingNullableToNonNullable = 8600` (has a memorable number)